### DISCUSSION TITLE
[Feature] Disable/Enable "Try it" stage dependent

### DISCUSSION THREAD

**Discussion by OllisGit** [Azure/api-management-developer-portal#2578] [Feature request]
Currently there is only a simple checkbox "Enable API Console" in the designer, but it would be really helpfull if its possible to assign a "condition" e.g. ```hostingUrl.startwith("https://production.myapi") == false```

Background:
We use two stages (2x API Management Service): production and qa. The API-Dev-Portal is not self-hosted!

In the qa-environment the developer should be able to try out the api-operations and call the qa-backends.
But in the prod-environment we want to disable this functionality.

Because we are using the migrate.js script to "transport" the "qa-dev-api design" to "prod-dev-api", one solution could be to "adjust" this script:
After the "capture" command of the migration-script is executed, the design-data is stored in the ```dist/snapshot/data.json``` file.
Then we can (depending on target-stage) do a search/replace of ```"enableConsole":true``` --> ```"enableConsole":false```

Is there maybe a better way to fulfill the requirement?

BR
Olli

**Comment by ghost** [comment]
@OllisGit, thank you for opening this issue. We will triage it within the next few business days.

**Comment by ghost** [comment]
@OllisGit, thank you for requesting this feature.

**Comment by mikebudzynski** [comment]
@OllisGit, thanks for opening the issue. The current workaround is to self-host the portal, where you can extend this functionality with custom code.
